pkg/eventprocessor/impl/executor/impl/executor_test.go

```diff
@@ -0,0 +1,265 @@
+package impl
+
+import (
+	"context"
+	"database/sql"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/stretchr/testify/require"
+	"github.com/textileio/go-tableland/internal/tableland"
+	"github.com/textileio/go-tableland/pkg/eventprocessor"
+	"github.com/textileio/go-tableland/pkg/eventprocessor/eventfeed"
+	"github.com/textileio/go-tableland/pkg/parsing"
+	parserimpl "github.com/textileio/go-tableland/pkg/parsing/impl"
+	"github.com/textileio/go-tableland/pkg/sqlstore/impl/system"
+	"github.com/textileio/go-tableland/pkg/tables/impl/ethereum"
+	"github.com/textileio/go-tableland/tests"
+)
+
+var chainID = 1337
+
+func TestReceiptExists(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	ex, _ := newExecutorWithTable(t, 0)
+
+	txnHash := "0x0000000000000000000000000000000000000000000000000000000000001234"
+
+	bs, err := ex.NewBlockScope(ctx, 0)
+	require.NoError(t, err)
+	ok, err := bs.TxnReceiptExists(ctx, common.HexToHash(txnHash))
+	require.NoError(t, err)
+	require.False(t, ok)
+	require.NoError(t, bs.Commit())
+	require.NoError(t, bs.Close())
+
+	bs, err = ex.NewBlockScope(ctx, 0)
+	require.NoError(t, err)
+	err = bs.SaveTxnReceipts(ctx, []eventprocessor.Receipt{
+		{
+			ChainID:     tableland.ChainID(chainID),
+			BlockNumber: 100,
+			TxnHash:     txnHash,
+		},
+	})
+	require.NoError(t, err)
+	require.NoError(t, bs.Commit())
+	require.NoError(t, bs.Close())
+
+	bs, err = ex.NewBlockScope(ctx, 0)
+	require.NoError(t, err)
+	ok, err = bs.TxnReceiptExists(ctx, common.HexToHash(txnHash))
+	require.NoError(t, err)
+	require.True(t, ok)
+	require.NoError(t, bs.Commit())
+	require.NoError(t, bs.Close())
+
+	require.NoError(t, ex.Close(ctx))
+}
+
+func TestMultiEventTxnBlock(t *testing.T) {
```

Comment:
This new test thoroughly tests a complex scenario with multiple txns in the same block, each with multiple transactions.

Note that the way the code is architected now, we can create e2e tests with this although the SC doesn't have any new method to create this situation from SC calls. 

Let's take a look here...